src/main/scala/cromwell/engine/package.scala

```diff
@@ -49,13 +49,48 @@ package object engine {
    * created
    */
   case class WorkflowDescriptor(id: WorkflowId, sourceFiles: WorkflowSourceFiles) {
+    // TODO: Extract this from here (there is no need to reload the configuration for each workflow)
+    // Not private because overridden in tests
+    lazy private [engine] val conf = ConfigFactory.load
+
     val workflowOptions = Try(sourceFiles.workflowOptionsJson.parseJson) match {
       case Success(options: JsObject) => WorkflowOptions.fromJsonObject(options).get // .get here to purposefully throw the exception
       case Success(other) => throw new Throwable(s"Expecting workflow options to be a JSON object, got $other")
       case Failure(ex) => throw ex
     }
 
-    val backend = Backend.from(workflowOptions.getOrElse("default_backend", ConfigFactory.load.getConfig("backend").getString("backend")))
+    val props = sys.props
+    val workflowLogger = props.get("LOG_MODE") match {
+      case Some(x) if x.toUpperCase.contains("SERVER") => makeFileLogger(
+        Paths.get(props.getOrElse("LOG_ROOT", ".")),
+        s"workflow.$id.log",
+        Level.toLevel(props.getOrElse("LOG_LEVEL", "debug"))
+      )
+      case _ => NOPLogger.NOP_LOGGER
+    }
+
+    // Call Caching
+    // TODO: Add to lenthall
+    def getConfigOption(key: String): Option[Config] = if (conf.hasPath(key)) Option(conf.getConfig(key)) else None
+    private lazy val configCallCaching = getConfigOption("call-caching") map { _.getBooleanOr("enabled", DefaultCallCachingValue) } getOrElse DefaultCallCachingValue
+    private lazy val optionCacheWriting = workflowOptions.getBoolean("write-to-cache") getOrElse configCallCaching
+    private lazy val optionCacheReading = workflowOptions.getBoolean("read-from-cache") getOrElse configCallCaching
+
+    if(!configCallCaching && optionCacheWriting) {
```

Comment:
Actually I think I'd prefer to see this as a nested if ...

    if (!configCallCaching) {
      if (optionalCacheWriting) { ... }
      if (optionalCacheReading) { ...}
    }